frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -55,13 +86,50 @@ export function DynamicConfiguration () {
         formSubmitting: false as boolean,
     });
 
+    let formErrors = {} as any;
+    const parseResult = formSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
+
     useEffect(() => {
         dispatch(listConfigurations({configScope: 'global', numVersions: 5}));
     }, [dispatch]);
 
+    useEffect(() => {
+        dispatch(listEMRApplications());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const appList: any[] = [];
+        for (const application of applicationList) {
+            const app = {
+                value: application,
+                label: application
+            };
+            appList.push(app);
+        }
+        // Make sure we're always starting from an empty array to prevent duplicates
+        setSelectedApplicationOptions([]);
+        for (const configApp of applicationConfig.configuration.EMRConfig.applications) {
```

Comment:
This could also probably be a map: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Not sure how I'd fit all this conditional logic and state change in a map. If you have suggested code I'd be happy to implement it, otherwise this code is functioning and this request is ultimately a preference.